timer.go

```diff
@@ -0,0 +1,38 @@
+package valkyrietry
+
+import "time"
+
+type Timer struct {
+	timer *time.Timer
```

Comment:
*thought* : This struct are being used as **convenience** helper such that when you call `timer.Start` it will lazily initialize internal `timer`

*question (non-blocking)* : is this abstraction really necessary ? Could we just use zero initialize `*time.Timer` instead then initialize it on the loop ? 

*suggestion* : you could also create helper method that checks whether a timer is nil then re-create new timer, this is perfectly safe I think.

Reply:
>  is this abstraction really necessary ? Could we just use zero initialize *time.Timer instead then initialize it on the loop ?

Actually, it's not strictly necessary; it's only used to wrap the `time.Timer` for implementing pauses between each retry.

> Could we just use zero initialize *time.Timer instead then initialize it on the loop ?

Ah, you're right. Perhaps we can directly initialize the `time.Timer` and then start or pause it as needed. I'll make that update.

Actually, the reason I created that abstraction was to have a helper function that wraps the timer, allowing for both initiation and reset in a single function.

Reply:
> Actually, the reason I created that abstraction was to have a helper function that wraps the timer, allowing for both initiation and reset in a single function.

Noted, this is very nitpick suggestion from me, though so this is non-blocking suggestion.